src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1468,6 +1468,27 @@ public static int Sign(float value)
             throw new ArithmeticException(SR.Arithmetic_NaN);
         }
 
+        // Helper functions for CopySign
+        // >= 0: returns 1
+        // < 0: returns -1
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int SignZeroToOne(int value)
+        {
+            return unchecked((value >> (32 - 1)) - (~value >> (32 - 1)));
```

Comment:
A better bit operation:
`(value >> (32 - 2)) | 1`

Also note that unchecked is the default option, so usually we don't specify it.

Reply:
> (value >> (32 - 2)) | 1

Your suggested expression is better indeed.

> Also note that unchecked is the default option, so usually we don't specify it.

I will remove the `unchecked` in these places, but keep it in the function body that has `checked`, so that it can emphasize and prevent other contributors from mistakenly adding `checked`.